clients/client_send_file.py

```diff
@@ -79,30 +82,47 @@ async def reconnected_cb():
         print(e)
         show_usage_and_die()
 
-    with open(data, mode="rb") as f:
-        byte_im = f.read()
-
     path, file = os.path.split(data)
     name, ext = file.split(".")
 
+    # Ask to upload an image
     req = ImageRequest()
     req.name = name
-    req.image = byte_im
+    req.confirmation = False
     payload = req.SerializeToString()
 
-    if args.request:
+    response = await nc.request(args.subject, payload, timeout=10)
+    print(f"Requested on [{args.subject}] : '{file}'")
+
+    res = ImageResponse.FromString(response.data)
+    print("Got response with presigned URL")
+
+    # Upload image with presigned URL
+    with open(data, mode="rb") as f:
+        byte_im = f.read()
+
+    res = requests.put(res.URL, data=byte_im)
+
+    if res.status_code == 200:
+        # Confirmation send
+        req = ImageRequest()
+        req.name = name
+        req.confirmation = True
+        payload = req.SerializeToString()
+
         response = await nc.request(args.subject, payload, timeout=10)
         print(f"Requested on [{args.subject}] : '{name}'")
 
-        res = ImageResponse.FromString(response.data)
+        # Modified images
+        res = ImageModifiedResponse.FromString(response.data)
```

Comment:
What if now the server wants to implement a new transformation? Do we have to update the message format every time? Isn't there a more elegant solution?